src/Modal/ModalLayer.jsx

```diff
@@ -52,29 +52,31 @@ function ModalLayer({
     };
   }, [isOpen]);
 
-  if (!isOpen) {
+  if (!isOpen && !isHidden) {
     return null;
   }
 
   const onClickOutside = !isBlocking ? onClose : null;
 
   return (
-    <ModalContextProvider onClose={onClose} isOpen={isOpen} isBlocking={isBlocking}>
+    <ModalContextProvider onClose={onClose} isOpen={isOpen} isHidden={isHidden} isBlocking={isBlocking}>
       <Portal>
         <FocusOn
           allowPinchZoom
           scrollLock
-          enabled={isOpen}
+          enabled={isOpen && !isHidden}
```

Comment:
`FocusOn` is used in `ModalDialog` to handle the accessibility of auto-focus on the first interactive element in the modal and handling the `Esc` keypress or backdrop click.